zstd.h

```diff
@@ -2530,3 +2531,7 @@ ZSTDLIB_API size_t ZSTD_insertBlock    (ZSTD_DCtx* dctx, const void* blockStart,
 #if defined (__cplusplus)
 }
 #endif
+#else /* USE_EXTERNAL_ZSTD */
+#undef ZSTD_STATIC_LINKING_ONLY
```

Comment:
This is a bit hacky to define `ZSTD_STATIC_LINKING_ONLY` in the Go file and then unset in the C file. Is there a way to just have the switch in `external_zstd.go` ? (i.e: an else that sets it)
(I don't remember if we need to define & import `zstd.h` in each go file)

Reply:
Are you sure we actually need to define ZSTD_STATIC_LINKING_ONLY? It appears that library is building fine when that's not defined.